src/modules/roc_packet/fec.h

```diff
@@ -21,8 +21,8 @@ namespace packet {
 
 //! FECFRAME packet.
 struct FEC {
-    //! Seqnum of first source packet in block.
-    seqnum_t blknum;
+    //! The source block number.
+    seqnum_t source_block_number;
```

Comment:
> I do not want to use seqnum_t as a type for the source block number. It will
  be more reasonable to define a new type for it. I am not sure where to put
  this stuff. I only found roc_packet/units.h. Should we also define types for
  other fec-specific packet fields: source_block_length?

We need seqnum_t to handle the overlap correctly. Its size should be the same as in the RTP packet.

Block number may overlap too and we need to handle its overlap as well. So I guess we need blknum_t. I think we need the following:

**units.h**

```
typedef uint32_t blknum_t;
typedef int32_t signed_blknum_t;
```

**fec.h**

```
struct FEC {
    //! The source block number.
    blknum_t source_block_number;

    //! The number of source packet per block.
    blknum_t source_block_length;

    ...
```

Reply:
To handle overlaps correctly, the length of the blknum_t should be smaller or equal to the length of the SBN in Payload ID header.

Reed-solomon uses 32-bit SBNs, but LDPC uses 16-bit SBNs, so we need 16-bit blknum.

Reply:
Done